rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+from rlberry.envs import Chain
+from rlberry.envs.benchmarks.ball_exploration import PBall2D
+from rlberry.manager import AgentManager
+import numpy as np
+from rlberry.seeding import set_external_seed
+import tempfile
+import os
+
+SEED = 42
+
+
+def check_agent_manager(Agent, continuous_state=False):
+    """
+    Check that the agent is compatible with :class:`~rlberry.manager.AgentManager`.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env_kwargs = {}
+    try:
+        agent = AgentManager(
+            Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+        )
+        agent.fit()
+    except Exception as exc:
+        raise RuntimeError("Agent not compatible with Agent Manager") from exc
+
+    return agent
+
+
+def check_agent_almost_equal(state, agent1, agent2, n_checks=5):
+    """
+    Check that two agents are almost equal in the sense that they give the same
+    actions n_checks times on a given state.
+
+
+    Parameters
+    ----------
+
+    state: state
+        environment state
+    agent1: rlberry agent
+        first agent to be compared
+    agent2: rlberry agent
+        second agent to be compared
+    n_checks : int, default=5
+        number of checks to do.
+    """
+    result = True
+    set_external_seed(SEED)
+    actions1 = []
+    for f in range(5):
+        # test on 5 actions
+        actions1.append(agent1.agent_handlers[0].policy(state))
+    set_external_seed(SEED)
+    for f in range(5):
+        # test on 5 actions
+        action2 = agent2.agent_handlers[0].policy(state)
```

Comment:
Here, we are assuming that the agents have a `policy` method, which is not the case for all agents. For instance, agents for POMDPs will not have a policy that is a function of an observation only, since they must take the past (or a hidden state) into account.

Suggested solution: instead of comparing the actions, compare the output of the `eval` method of the agents.

Reply:
Ok, but at the same time there can be some problem with eval, for instance the user coded his agent with eval always returning 0, this would work. Or for bandits agent in particular, eval is just compare the optimal mean but two bandit agents can be different but still have the same optimal action (optimal arm identified). 
Is there a way to check without using eval ?

Reply:
So we can have different checks according to the type of the agent:
* If `AgentWithSimplePolicy`, we keep the test as it is; 
* If not (only `Agent` is implemented), then we use `eval`.

In general (for RL), the output of `eval` will depend on the policy implemented/followed by the agent, so it should be good in most cases.

Reply:
Or we can have a parameter that say `compare_with="eval"` or `compare_with="policy"` or any string that correspond to the name of a method that returns a real number, would that be alright ?

The solution with `AgentWithSimplePolicy` vs `Agent` is also alright but the solution with compare_with would allow maybe more generalization.

Reply:
Yes, I think that `compare_with` is a nice solution, since it doesn't hide from the user the criterion that we use to compare the agents!

Reply:
yes but the new code allow even more generality ! you can define a method like "agent.evaluation_with_weird_name" and use it in compare with, not only policy and eval !

Reply:
I checked eval in the test of the check (`test_check.py` file)